engine/access/state_stream/event.go

```diff
@@ -0,0 +1,56 @@
+package state_stream
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type ParsedEventType int
+
+const (
+	ProtocolEventType ParsedEventType = iota + 1
+	AccountEventType
+)
+
+type ParsedEvent struct {
+	Type      ParsedEventType
+	EventType flow.EventType
+	Address   string
+	Contract  string
+	Name      string
+}
+
+// ParseEvent parses an event type into its parts. There are 2 valid EventType formats:
+// - flow.[EventName]
+// - A.[Address].[Contract].[EventName]
+// Any other format results in an error.
+func ParseEvent(eventType flow.EventType) (*ParsedEvent, error) {
+	parts := strings.Split(string(eventType), ".")
+
+	switch parts[0] {
+	case "flow":
+		if len(parts) == 2 {
+			return &ParsedEvent{
+				Type:      ProtocolEventType,
+				EventType: eventType,
+				Contract:  parts[0],
+				Name:      parts[1],
+			}, nil
+		}
+
+	case "A":
+		if len(parts) == 4 {
+			return &ParsedEvent{
+				Type:      AccountEventType,
+				EventType: eventType,
+				Address:   parts[1],
+				Contract:  parts[2],
```

Comment:
This is incorrect. Contract should be the full contract. This implementation also needs a `ContractName` field to hold just `parts[2]`
```suggestion
				Contract:  fmt.Sprintf("%s.%s.%s", parts[0], parts[1], parts[2]),
```